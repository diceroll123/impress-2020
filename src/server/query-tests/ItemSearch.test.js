const gql = require("graphql-tag");
const { query, getDbCalls } = require("./setup.js");

describe("ItemSearch", () => {
  it("loads Neopian Times items", async () => {
    const res = await query({
      query: gql`
        query {
          itemSearch(query: "Neopian Times") {
            query
            items {
              id
              name
            }
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    expect(res.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchInlineSnapshot(`
      Array [
        Array [
          "SELECT items.*, t.name FROM items
               INNER JOIN item_translations t ON t.item_id = items.id
               WHERE t.name LIKE ? AND t.locale=\\"en\\"
               ORDER BY t.name
               LIMIT 30",
          Array [
            "%Neopian Times%",
          ],
        ],
      ]
    `);
  });

  it("loads Neopian Times items that fit the Starry Zafara", async () => {
    const res = await query({
      query: gql`
        query {
          itemSearchToFit(
            query: "Neopian Times"
            speciesId: "54"
            colorId: "75"
          ) {
            query
            items {
              id
              name
            }
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    expect(res.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchInlineSnapshot(`
      Array [
        Array [
          "SELECT * FROM pet_types WHERE (species_id = ? AND color_id = ?)",
          Array [
            "54",
            "75",
          ],
        ],
        Array [
          "SELECT DISTINCT items.*, t.name FROM items
                 INNER JOIN item_translations t ON t.item_id = items.id
                 INNER JOIN parents_swf_assets rel
                     ON rel.parent_type = \\"Item\\" AND rel.parent_id = items.id
                 INNER JOIN swf_assets ON rel.swf_asset_id = swf_assets.id
                 WHERE t.name LIKE ? AND t.locale=\\"en\\" AND
                     (swf_assets.body_id = ? OR swf_assets.body_id = 0)
                 ORDER BY t.name
                 LIMIT ? OFFSET ?",
          Array [
            "%Neopian Times%",
            "180",
            30,
            0,
          ],
        ],
      ]
    `);
  });

  it("loads the first 10 hats that fit the Starry Zafara", async () => {
    const res = await query({
      query: gql`
        query {
          itemSearchToFit(
            query: "hat"
            speciesId: "54"
            colorId: "75"
            offset: 0
            limit: 10
          ) {
            query
            items {
              id
              name
            }
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    expect(res.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchInlineSnapshot(`
      Array [
        Array [
          "SELECT * FROM pet_types WHERE (species_id = ? AND color_id = ?)",
          Array [
            "54",
            "75",
          ],
        ],
        Array [
          "SELECT DISTINCT items.*, t.name FROM items
                 INNER JOIN item_translations t ON t.item_id = items.id
                 INNER JOIN parents_swf_assets rel
                     ON rel.parent_type = \\"Item\\" AND rel.parent_id = items.id
                 INNER JOIN swf_assets ON rel.swf_asset_id = swf_assets.id
                 WHERE t.name LIKE ? AND t.locale=\\"en\\" AND
                     (swf_assets.body_id = ? OR swf_assets.body_id = 0)
                 ORDER BY t.name
                 LIMIT ? OFFSET ?",
          Array [
            "%hat%",
            "180",
            10,
            0,
          ],
        ],
      ]
    `);
  });

  it("loads the next 10 hats that fit the Starry Zafara", async () => {
    const res = await query({
      query: gql`
        query {
          itemSearchToFit(
            query: "hat"
            speciesId: "54"
            colorId: "75"
            offset: 10
            limit: 10
          ) {
            query
            items {
              id
              name
            }
          }
        }
      `,
    });

    expect(res).toHaveNoErrors();
    expect(res.data).toMatchSnapshot();
    expect(getDbCalls()).toMatchInlineSnapshot(`
      Array [
        Array [
          "SELECT * FROM pet_types WHERE (species_id = ? AND color_id = ?)",
          Array [
            "54",
            "75",
          ],
        ],
        Array [
          "SELECT DISTINCT items.*, t.name FROM items
                 INNER JOIN item_translations t ON t.item_id = items.id
                 INNER JOIN parents_swf_assets rel
                     ON rel.parent_type = \\"Item\\" AND rel.parent_id = items.id
                 INNER JOIN swf_assets ON rel.swf_asset_id = swf_assets.id
                 WHERE t.name LIKE ? AND t.locale=\\"en\\" AND
                     (swf_assets.body_id = ? OR swf_assets.body_id = 0)
                 ORDER BY t.name
                 LIMIT ? OFFSET ?",
          Array [
            "%hat%",
            "180",
            10,
            10,
          ],
        ],
      ]
    `);
  });
});
