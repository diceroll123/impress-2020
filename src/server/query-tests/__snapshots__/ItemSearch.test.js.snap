// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemSearch loads Neopian Times items 1`] = `
Object {
  "itemSearch": Object {
    "items": Array [
      Object {
        "id": "40431",
        "name": "Neopian Times Background",
      },
      Object {
        "id": "59391",
        "name": "Neopian Times Eyrie Hat",
      },
      Object {
        "id": "59392",
        "name": "Neopian Times Eyrie Shirt and Vest",
      },
      Object {
        "id": "59394",
        "name": "Neopian Times Eyrie Shoes",
      },
      Object {
        "id": "59393",
        "name": "Neopian Times Eyrie Trousers",
      },
      Object {
        "id": "59390",
        "name": "Neopian Times Eyries Paper",
      },
      Object {
        "id": "51098",
        "name": "Neopian Times Writing Quill",
      },
      Object {
        "id": "61101",
        "name": "Neopian Times Zafara Handkerchief",
      },
      Object {
        "id": "61100",
        "name": "Neopian Times Zafara Hat",
      },
      Object {
        "id": "61102",
        "name": "Neopian Times Zafara Shirt and Vest",
      },
      Object {
        "id": "61104",
        "name": "Neopian Times Zafara Shoes",
      },
      Object {
        "id": "61103",
        "name": "Neopian Times Zafara Trousers",
      },
    ],
    "query": "Neopian Times",
  },
}
`;

exports[`ItemSearch loads Neopian Times items that fit the Starry Zafara 1`] = `
Object {
  "itemSearchToFit": Object {
    "items": Array [
      Object {
        "id": "40431",
        "name": "Neopian Times Background",
      },
      Object {
        "id": "51098",
        "name": "Neopian Times Writing Quill",
      },
      Object {
        "id": "61101",
        "name": "Neopian Times Zafara Handkerchief",
      },
      Object {
        "id": "61100",
        "name": "Neopian Times Zafara Hat",
      },
      Object {
        "id": "61102",
        "name": "Neopian Times Zafara Shirt and Vest",
      },
      Object {
        "id": "61104",
        "name": "Neopian Times Zafara Shoes",
      },
      Object {
        "id": "61103",
        "name": "Neopian Times Zafara Trousers",
      },
    ],
    "query": "Neopian Times",
  },
}
`;

exports[`ItemSearch loads the first 10 hats that fit the Starry Zafara 1`] = `
Object {
  "itemSearchToFit": Object {
    "items": Array [
      Object {
        "id": "74967",
        "name": "17th Birthday Party Hat",
      },
      Object {
        "id": "49026",
        "name": "Abominable Snowman Hat",
      },
      Object {
        "id": "67242",
        "name": "Accessories Shop Wig and Hat",
      },
      Object {
        "id": "64177",
        "name": "Acorn Hat",
      },
      Object {
        "id": "69995",
        "name": "Adventure in Pastel Hat and Wig",
      },
      Object {
        "id": "62375",
        "name": "Altador Cup Trophy Hat",
      },
      Object {
        "id": "56654",
        "name": "Altador Team Hat",
      },
      Object {
        "id": "62322",
        "name": "Altador Team Jester Hat",
      },
      Object {
        "id": "58733",
        "name": "Apple Bobbing Bart Hat",
      },
      Object {
        "id": "80401",
        "name": "Aurricks Finest Hat",
      },
    ],
    "query": "hat",
  },
}
`;

exports[`ItemSearch loads the next 10 hats that fit the Starry Zafara 1`] = `
Object {
  "itemSearchToFit": Object {
    "items": Array [
      Object {
        "id": "50168",
        "name": "Babaa Hat",
      },
      Object {
        "id": "78311",
        "name": "Backwards Hat and Wig",
      },
      Object {
        "id": "66653",
        "name": "Bagel Hat Wig",
      },
      Object {
        "id": "51366",
        "name": "Balloon Sculpture Hat",
      },
      Object {
        "id": "66236",
        "name": "Banker Hat and Wig",
      },
      Object {
        "id": "52733",
        "name": "Battlefield Legends Hat",
      },
      Object {
        "id": "57617",
        "name": "Big Black Lace Pirate Hat",
      },
      Object {
        "id": "44451",
        "name": "Big Fuzzy Hat",
      },
      Object {
        "id": "58318",
        "name": "Bigsby Shadingtons Hat",
      },
      Object {
        "id": "39883",
        "name": "Bird Nest Hat",
      },
    ],
    "query": "hat",
  },
}
`;
